packages/calypso-e2e/src/lib/flows/site-assembler-flow.ts

```diff
@@ -40,6 +40,28 @@ export class SiteAssemblerFlow {
 	 * @param {number} index Index of the item to choose. Defaults to 0.
 	 */
 	async selectLayoutComponent( index = 0 ): Promise< void > {
-		await this.page.locator( '.pattern-list-renderer__pattern-list-item' ).nth( index ).click();
+		const target = this.page
+			.getByRole( 'listbox', { name: 'Block patterns' } )
+			.getByRole( 'option' )
+			.nth( index )
+			.locator( 'iframe' );
+
+		// The iframe has an attribute of "height" that is present only if the preview has loaded.
+		// By checking for the presence of the "height" attribute on the last visible
+		// iframe, we can be sure that all visible component card has loaded.
+		// If the last iframe does not yet contain the "height" attribute, then wait
+		// for a predetermined timeout, then re-evaluate.
+		while (
+			! ( await target.evaluate( ( element ) => {
+				if ( element.style.height ) {
+					return true;
+				}
+				return false;
+			} ) )
+		) {
+			await this.page.waitForTimeout( 1000 );
+		}
+
+		await target.click();
```

Comment:
What if we pierce through the iframe, forcing it to have the inner target ready instead? For example, I think something like this might work:

```js
async selectLayoutComponent( index = 0 ): Promise< void > {
	await this.page
		.getByRole( 'listbox', { name: 'Block patterns' } )
		.getByRole( 'option' )
		.nth( index )
		.frameLocator( 'iframe' )
		.locator( '.block-editor-iframe__body' )
		.click();
}

```

Reply:
> What if we pierce through the iframe, forcing it to have the inner target ready instead? For example, I think something like this might work:
> 
> ```js
> async selectLayoutComponent( index = 0 ): Promise< void > {
> 	await this.page
> 		.getByRole( 'listbox', { name: 'Block patterns' } )
> 		.getByRole( 'option' )
> 		.nth( index )
> 		.frameLocator( 'iframe' )
> 		.locator( '.block-editor-iframe__body' )
> 		.click();
> }
> ```

Blargh, you come in with a good suggestion right after I hit merge! 

I'm going to quickly try your idea in another PR.